inventory/inventory.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+
+"""
+Environment variable for this inventory script:
+OL_TYPE:        required, the openlab deployment type, one of: allinone, allinone-ha, openlab, openlab-ha
+OL_{host}_IP:   new ip address specified for the {host}, such as: OL_ZUUL01_IP, OL_NODEPOOL01_IP,
+                the {host} must be one of the host name defined in the yaml inventory files.
+OL_SWITCH_MASTER_SLAVE: whether switch the master/slave IP address
+
+"""
+import os
+import sys
+
+import argparse
+import json
+import subprocess
+
+
+def parse_inventory():
+    inventory_file = 'inventory/%s.yaml' % os.environ.get('OL_TYPE')
+    parsed_inventories = subprocess.Popen(
+        ['ansible-inventory', '-i', inventory_file, '--list'],
+        stdout=subprocess.PIPE).stdout.read()
+    updated_inventories = json.loads(parsed_inventories)
+    inventory_hosts = updated_inventories['_meta']['hostvars']
+    host_names = inventory_hosts.keys()
+    host_names.remove('bastion')
+    ips_update = dict([(h, os.environ.get("OL_%s_IP" % h.upper())) for h in host_names])
+
+    old_ips = dict([(h, inventory_hosts[h]['ansible_host']) for h in host_names])
+
+    switch_role = os.environ.get('OL_SWITCH_MASTER_SLAVE')
+
+    if switch_role:
+        for host in host_names:
+            if host.endswith('01') and host[:-2] in ['zuul', 'nodepool']:
```

Comment:
The host name must be end with 01/02 ?

Reply:
it have to distinguish the slave hosts and master hosts, I am not sure if there is better choice, and this is work with the current yaml files under inventory/

Reply:
We have a hostname format `{cloud-provider}-openlab-{type}` in https://github.com/theopenlab/labkeeper/blob/master/openlabcmd/openlabcmd/cli.py#L84 , like:
1. citynetwork-openlab-zuul
2. citynetwork-openlab-nodepool
3. vexxhost-openlab-zuul
4. vexxhost-openlab-nodepool
5. huaweicloud-openlab-zookeeper

Just switching two nodes with same suffix

Reply:
this host name is not the `hostname` of the VMs,  and it is unrelated with the hostname of VMs, here we only care about the IP address and it's role. it is says, here the xxx01, xxx02 only used by labkeeper, and do not care about the cloud provider name.